app/todos/resources.py

```diff
@@ -0,0 +1,153 @@
+from flask import request, session
+from flask_restful import Resource
+from sqlalchemy import and_
+
+from app import db, Todo, TodoListPermission
+from app.lib import authentication, response_util, status
+
+
+class TodoMultiResource(Resource):
+    """Class for creating and accessing todos.
+    """
+    @authentication.requiresAuth
+    def get(self):
+        """Method gets a list of all the todos and returns them in an OK
+        response.
+        """
+        userId = session.get('userId')
+        permissions = TodoListPermission.query.filter_by(userId=userId)
+        todoListId = request.args.get('todoListId')
+
+        if todoListId is None:
+
+            todoListIds = [permission.todoListId for permission in permissions]
+            if not todoListIds:
+                return response_util.buildOkResponse([])
+
+            myTodos = Todo.query.filter(Todo.todoListId.in_(todoListIds))
+
+            return response_util.buildOkResponse([todo.toDict() for todo in myTodos])
+        else:
+            todoListId = int(todoListId)  # bad request if not int
+            permissions = TodoListPermission.query.filter(
+                           and_(TodoListPermission.todoListId == todoListId,
+                                TodoListPermission.userId == userId)).all()
+
+            if not permissions:
+                raise status.Forbidden()
+
+            todos = Todo.query.filter_by(todoListId=todoListId)
+
+            return response_util.buildOkResponse([todo.toDict() for todo in todos])
+
+    @authentication.requiresAuth
+    def post(self):
+        """Method adds a new todo.
+        """
+        userId = session.get('userId')
+        todoListId = request.form.get('todoListId')
+
+        if not(request.form.get('subject') and request.form.get('todoListId')):
```

Comment:
You can replace request.form.get('todoListId') with the variable todoListId that you defined above.